clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/provider/controller/CacheArtifactController.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 Mirantis, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.provider.controller;
+
+import com.netflix.spinnaker.cats.cache.Cache;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.clouddriver.provider.cache.Keys;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.netflix.spinnaker.clouddriver.provider.cache.Keys.Namespace.HELM_NAMES;
+import static com.netflix.spinnaker.clouddriver.provider.cache.Keys.Namespace.HELM_VERSIONS;
+
+
+@RestController
+@RequestMapping("/artifacts")
+class CacheArtifactController {
+
+  Cache cacheView;
+
+  @Autowired
+  CacheArtifactController(Cache cacheView) {
+    this.cacheView = cacheView;
+  }
+
+
+  @RequestMapping(method = RequestMethod.GET, value = "{type}/account/{accountName}/names")
+  List<String> getNames(@PathVariable("type") String type,
+                        @PathVariable("accountName") String accountName) {
+    switch (type) {
+      case "helm":
+        return findAttributeValue(HELM_NAMES.ns, Keys.getHelmNameKey(accountName), "names");
+      default:
+        throw new NotFoundException("Could not found artifact of type " + type);
+    }
+  }
+
+  @RequestMapping(method = RequestMethod.GET, value = "{type}/account/{accountName}/names/{artifactName}/versions")
```

Comment:
Usually these `@RequestMapping`s are prefixed with a `/`; I'm actually not 100% sure what would happen if a nested one (as this) didn't start with a slash (maybe it's implied)...but it's probably clearer to include it.